AppController/djinn.rb

```diff
@@ -3535,12 +3535,14 @@ def start_datastore
 
       assignments['datastore'] = {'count' => server_count,
                                   'verbose' => verbose}
+      ZKInterface.set_machine_assignments(node.private_ip, assignments)
     }
-    ZKInterface.set_machine_assignments(my_node.private_ip, assignments)
 
-    # Let's wait for at least one datastore server to be active.
-    until HelperFunctions.is_port_open?(db_proxy, DatastoreServer::PROXY_PORT)
-      update_db_haproxy if my_node.is_load_balancer?
+    # Let's wait for at least one datastore server to be active. Since
+    # only the Shadow assign Datastore, we can safely check locally when
+    # haproxy is setup.
+    until HelperFunctions.is_port_open?(@my__private_ip, DatastoreServer::PROXY_PORT)
```

Comment:
I think it should still be checking the port on the load balancer since that what datastore clients use.

Reply:
The shadow is one load balancer for sure at this time, so in a normal enviroment, that's it. In HA, I'm not sure I would like to wait for the secondary LB to come up: what if there is an issue? If shadow is running, we are guaranteed that one LB is active, and it is indeed the working one.

Reply:
The shadow role is conceptually different than a load balancer. The fact that they happen to coexist on the same machine right now is not a detail we should rely upon. The process of checking if the service is available should be checking whether it is available to clients, which use a load balancer.